pkg/upgrade/upgrader/exec.go

```diff
@@ -122,9 +124,10 @@ func Exec(fromVersion, toVersion, kind, name,
 	}
 
 	if err != nil {
-		if utaskObj != nil {
+		if utaskObj != nil && isENVPresent {
 			backoffLimit, uerr := getBackoffLimit(openebsNamespace)
 			if uerr != nil {
+				klog.Error(err)
```

Comment:
I think `err` should be returned and uerr should be logged

Reply:
Makes sense. Done